Sources/_SwiftFormatTestSupport/Parsing.swift

```diff
@@ -0,0 +1,34 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+@_spi(ExperimentalLanguageFeatures) import SwiftParser
+import SwiftSyntax
+import XCTest
+
+/// Parses the given source string and returns the corresponding `SourceFileSyntax` node.
+///
+/// - Parameters:
+///   - source: The source text to parse.
+///   - experimentalFeatures: The set of experimental features that should be enabled in the parser.
+@_spi(Testing)
+public func parseForTesting(
+  source: String,
+  experimentalFeatures: Parser.ExperimentalFeatures
+) -> SourceFileSyntax {
```

Comment:
Do we need this still? I think there’s a `Parser.parse` function that does exactly this now that we don’t need to map experimental features anymore.

Reply:
Since the `Parser.parse` overload that takes experimental features only takes a UTF-8 pointer, I mainly kept it to tidy up the handful of call sites in tests where we would otherwise have to repeat the "make a mutable copy of the string, convert it to a UTF-8 pointer, and then parse it" dance.

Reply:
Oh, I see. I would make it an extension of `Parser` then.

Reply:
Done.